content/blog/2018/06/2018-06-19-new-design-digest.adoc

```diff
@@ -0,0 +1,162 @@
+  <j:new var="h" className="hudson.Functions"/>
+  <html>
+    <head>
+      <!-- css styling, will fallback to default implementation -->
+      <st:include it="${h.simpleDecorator}" page="simple-head.jelly" optional="true"/>
+    </head>
+    <body>
+      <div class="simple-page" role="main">
+        <st:include it="${h.simpleDecorator}" page="simple-header.jelly" optional="true"/>
+      </div>
+      <div class="footer">
+        <st:include it="${h.simpleDecorator}" page="simple-footer.jelly" optional="true"/>
+      </div>
+    </body>
+</j:jelly>
+----
+
+The *sign-up* page only supports the simple-head.jelly:
+
+[source,html]
+----
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" >
+  <j:new var="h" className="hudson.Functions"/>
+  <html>
+    <head>
+      <!-- css styling, will fallback to default implementation -->
+      <st:include it="${h.simpleDecorator}" page="simple-head.jelly" optional="true"/>
+    </head>
+</j:jelly>
+----
+
+== SimplePageDecorator - custom implementations
+
+NOTE: Have a look at https://github.com/daniel-beck/login-theme-plugin[login-theme-plugin], which allows you to
+configure your own custom content to be injected into the new login/sign-up page
+
+To allow easy customisation the decorator only implements one instance by the principal "first-come-first-serve".
+If jenkins finds an extension of the SimplePageDecorator it will use the jelly files provided by that plugin if
+not Jenkins will fallback to the default implementation provided by core.
+
+[source,java]
+----
+@Extension
+public class MyLoginPageDecorator extends LoginPageDecorator {
+   public String getProductName() {
+     return "MyJenkins";
+   }
+}
+----
+
+As soon as a plugin implements the above as a java extension, all the provided *jelly* files will get preference before
+the default implementation.
+
+To customize the logo we display in the login page and create a *simple-head.jelly* like:
+
+[source,html]
+----
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+  <link rel="stylesheet" href="${resURL}/css/simple-page.css" type="text/css" />
+  <link rel="stylesheet" href="${resURL}/css/simple-page.theme.css" type="text/css" />
+  <style>
+    .simple-page .logo {
+        background-image: url('${resURL}/plugin/YOUR_PLUGIN/icons/my.svg');
+        background-repeat: no-repeat;
+        background-position: 50% 0;
+        height: 130px;
+    }
+  </style>
+  <link rel="stylesheet" href="${resURL}/css/simple-page-forms.css" type="text/css" />
+</j:jelly>
+----
+
+To customize further the login page we can inject e.g. a tagline, create a *simple-header.jelly* like:
+
+[source,html]
+----
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+    <div id="loginIntro">
+        <div class="logo"> </div>
+        <h1 id="productName">Welcome to ${it.productName}!</h1>
+    </div>
+</j:jelly>
+----
+
+The above technique is used to create:
```

Comment:
Would it be possible to replace the vendor screenshot by something neutral?
Otherwise it may make impression that this vendor is pushing such breaking change just to address its needs.

It's not correct in this case AFAIK

Reply:
Well, IMHO it is a bit of "marketing" for the sponsoring company and if you want I will at the disclaimer: "The following screenshot is from the custom theme of the sponsoring unit of the change" I mean that is simply stating the facts and really open.